web-components/src/components/robotoff/robotoff-question-form.ts

```diff
@@ -0,0 +1,163 @@
+import { ButtonType, getButtonClasses } from "../../styles/buttons"
+import { EventType } from "../../constants"
+import { classMap } from "lit/directives/class-map.js"
+import { answerQuestion } from "../../signals/questions"
+/**
+ * RobotoffQuestionForm component
+ * It displays a form to answer a question about a product.
+ */
+@customElement("robotoff-question-form")
+export class RobotoffQuestionForm extends LitElement {
+  static override styles = [
+    ...getButtonClasses([
+      ButtonType.Cappucino,
+      ButtonType.White,
+      ButtonType.LINK,
+    ]),
+    css`
+      :host {
+        display: block;
+        max-width: 800px;
+      }
+
+      .question-form {
+        display: flex;
+        align-items: center;
+        flex-direction: column;
+        gap: 1rem;
+      }
+
+      .question-img-wrapper {
+        position: relative;
+        justify-content: center;
+        width: 100px;
+      }
+
+      .question-img-wrapper.enlarged {
+        width: 100%;
+        max-width: 400px;
+      }
+
+      .question-img-wrapper img {
+        width: 100%;
+      }
+
+      .img-button-wrapper {
+        position: absolute;
+        bottom: 0.5rem;
+        right: 0.5rem;
+        display: flex;
+        justify-content: center;
+      }
+    `,
+  ]
+
+  /**
+   * The question to display.
+   */
+  @property({ type: Object, reflect: true })
+  question?: Question
+
+  /**
+   * The image size is zoomed or not.
+   */
+  @state()
+  private _zoomed: boolean = false
+
+  /**
+   * Emit an event submit when the user clicks on a button.
+   * It stops the propagation of the event to avoid the click event on the parent.
+   */
+  private emitEventClick = (event: Event, value: string) => {
+    event.stopPropagation()
+    const click = new CustomEvent(EventType.SUBMIT, {
+      detail: { value },
+      bubbles: true,
+      composed: true,
+    })
+
+    this.dispatchEvent(click)
+  }
+  private _annotateProduct = async (
+    event: Event,
+    value: QuestionAnnotationAnswer
+  ) => {
+    answerQuestion(this.question?.insight_id!, value)
+    this.emitEventClick(event, value)
+  }
+
+  private _toggleImageSize() {
+    this._zoomed = !this._zoomed
+  }
+
+  private _renderImage() {
+    if (!this.question?.source_image_url) {
+      return nothing
+    }
+
+    return html`<div>
+      <div
+        class=${classMap({
+          "question-img-wrapper": true,
+          enlarged: this._zoomed,
+        })}
+      >
+        <div>
+          <img .src=${this.question?.source_image_url} alt="Product image" />
+        </div>
+
+        <div class="img-button-wrapper">
+          <zoom-unzoom-button
+            .zoomed=${this._zoomed}
+            @click=${this._toggleImageSize}
+          ></zoom-unzoom-button>
+        </div>
+      </div>
+    </div> `
+  }
+
+  override render() {
+    if (!this.question) {
+      return html`<div>No question</div>`
+    }
```

Comment:
We will have to see, but I think we don't want to display the component at all in this case.

Reply:
Ok, I saw below that you indeed don't display the whole robotoff-question in this case, but that's a good idea to have a fallback for the form in case it appears for several ms.